Introduce support for tag synonymy (#21)
informalsystems/vdd: traceability/traceability.md

  programmers. There is no danger of confusing a Rust package name with a tag,
  as the tags always come with revision numbers.

|TRC-TAG.1::SYNTAX.1::SYNONYMY.1| |TRC-IMPL.1::SYNONYMY.1|
: A logical unit may bear multiple tags, as a result of implementing multiple
  ancestor logical units. This gives rise to "synonyms tags". However, logical

>   units with synonymous tags should always have a unique leaf tag.

Author: ```suggestion
  units with synonymous tags should always terminate with the same leaf tag.
```

>   A logical unit is tagged with synonymous tags by separating the synonymous

Author: ```suggestion
  A logical unit is tagged with synonymous tags by separating the synonyms
```

  with a space. Given a unit with the leaf tag `T` that implements `n` ancestor
  units with paths `path[0]...path[n]`, the unit is tagged with `|path[0]::T|
  |path[1]::T| ... |path[n]::T|`.

Example: The previous logical unit implements both [TRC-TAG.1::SYNTAX.1] and
[TRC-IMPL.1]. As a result, it is tagged with both
[TRC-TAG.1::SYNTAX.1::SYNONYMY.1] and [TRC-IMPL.1::SYNONYMY.1], and so these two

> tags are synonymous. However, it also bears the univocal leaf tag `SYNONYMY.1`

Author: ```suggestion
tags are synonymous. However, all its tags end with the single leaf tag `SYNONYMY.1`
```
